### PR TITLE
TIKATEXTEXTRACT - Translated to typescirpt

### PR DESCRIPTION
**Pull request by whodeen**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,16 +31,23 @@
...
+    "@babel/core": "7.14.3",
+    "@babel/preset-env": "7.14.2",
+    "@shelf/babel-config": "0.1.2",
+    "@shelf/eslint-config": "1.1.0",
+    "@shelf/prettier-config": "1.0.0",
+    "@shelf/tsconfig": "^0.0.6",
+    "@types/jest": "26.0.23",
+    "eslint": "7.27.0",
     "eslint-config-google": "0.6.0",
     "eslint-formatter-pretty": "^1.3.0",
```
Comment (vladgolubev) [inline package.json:44]: Not needed anymore

File: package.json [hunk 2]
```diff
@@ -27,16 +31,23 @@
     "is-stream": "^1.1.0"
   },
   "devDependencies": {
-    "babel-cli": "^6.26.0",
-    "babel-jest": "^22.0.0",
-    "babel-preset-env": "^1.6.1",
-    "eslint": "^4.13.1",
+    "@babel/cli": "7.14.3",
+    "@babel/core": "7.14.3",
+    "@babel/preset-env": "7.14.2",
```
Comment (vladgolubev) [inline package.json:36]: Not needed anymore, it's already part of `@shelf/babel-config`

File: package.json [hunk 3]
```diff
@@ -27,16 +31,23 @@
...
-    "memorystream": "^0.3.1"
+    "jest": "27.0.1",
+    "memorystream": "^0.3.1",
+    "prettier": "2.3.0",
+    "typescript": "4.2.4"
   },
-  "jest": {
-    "testEnvironment": "node"
+  "engines": {
+    "node": ">=6"
```
Comment (vladgolubev) [inline package.json:51]: Let's set it to Node 14 now

File: src/index.ts [hunk 4]
```diff
@@ -3,5 +3,5 @@ import {extract} from './text';
 
 module.exports = {
```
Comment (vladgolubev) [inline src/index.ts:4]: Since this is typescript file, let's export these functions in a typescript way `export ...`

**Comment by vladgolubev** [comment]
Please change Node.js version in CircleCI config to 14 as well

File: src/index.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import {startServer} from './server';
 import {extract} from './text';
```
Comment (vladgolubev) [inline src/index.ts:2]:
```suggestion
export {startServer} from './server';
export {extract} from './text';
```

File: package.json [hunk 6]
```diff
@@ -2,23 +2,27 @@
   "name": "tika-text-extract",
```
Comment (vladgolubev) [inline package.json:2]:
```suggestion
  "name": "@shelf/tika-text-extract",
```

It was previously under my personal user. Let's make it on behalf of Shelf

File: package.json [hunk 7]
```diff
@@ -2,23 +2,27 @@
...
   "version": "2.4.0",
   "description": "Extract text from a document by Apache Tika",
+  "repository": "vladgolubev/tika-text-extract",
```
Comment (vladgolubev) [inline package.json:5]:
```suggestion
  "repository": "shelfio/tika-text-extract",
```

File: package.json [hunk 8]
```diff
@@ -2,23 +2,27 @@
   "name": "tika-text-extract",
   "version": "2.4.0",
   "description": "Extract text from a document by Apache Tika",
```
Comment (vladgolubev) [inline package.json:4]:
```suggestion
  "description": "Extract text from documents with Apache Tika",
```

File: src/index.ts (hunk 5 above)
Comment (whodeen) [reply src/index.ts:2]: that's actually was my initial idea, but I thought it's no good since it's changing the way it needs to be imported further

**Review by vladgolubev — APPROVED** [review]

**Pull request merged**